contrib/mypy/src/python/pants/contrib/mypy/register.py

```diff
@@ -11,4 +11,4 @@
 
 
 def register_goals():
-  task(name='mypy', action=MypyTask).install('mypy')
+  task(name='mypy', action=MypyTask).install('lint')
```

Comment:
I'm on the fence about this one. Should running `mypy` act like "compilation", in that it happens before pytest runs? Would it be better to get all of my tests passing and then see mypy errors, or vice versa?

In my mind the location of a task like this is all about how early the errors are useful to you: if seeing that types are incorrect would influence how you write tests, that maybe in `compile` or at the end of `pyprep` would make sense?

Reply:
Good question. `mypy`'s purpose is to find semantic errors, so I agree seems more like "compilation" and not pure-style like `pep8` then. A similar issue arises for the ESLint linter used in the contrib node plugin. (See https://eslint.org/docs/rules/ for examples of many semantic rules.)

Assuming there are type annotations and everything is configured properly, I'd probably want to see `mypy` errors before running tests. Just like I'd want to see any errors that ESLint or the Google Closure compiler may output for Javascript.

Maybe we should have a single answer for all such semantic checking tools?

Reply:
In my mind, a true semantic error equates to a guaranteed programming error and should always be part of compile. It's true that a typecheck pass could reject a program that works, but then either the types signatures specified incorrectly or the values passed or returned were luckily shaped correctly. A linter, on the other hand, generally includes checks that only enforce opinion - import sorting, line lengths, etc. It's true that some linters include semantic checks making those tools tricky to assign a goal, but for mypy, which is narrowly a type-checker, it seems to me the compile goal is the clear home. I think its probably fairly universal to associate "compile" with typechecking and generation of object code of whch mypy does the former.

Reply:
+1.